src/pipecat/utils/whatsapp/api.py

```diff
@@ -0,0 +1,345 @@
+        calls: List of call connection events
+    """
+
+    messaging_product: str
+    metadata: WhatsAppMetadata
+    contacts: List[WhatsAppContact]
+    calls: List[WhatsAppConnectCall]
+
+
+class WhatsAppTerminateCallValue(BaseModel):
+    """Webhook payload for call termination events.
+
+    Attributes:
+        messaging_product: Always "whatsapp"
+        metadata: Business phone number information
+        calls: List of call termination events
+        errors: Optional list of errors that occurred during the call
+    """
+
+    messaging_product: str
+    metadata: WhatsAppMetadata
+    calls: List[WhatsAppTerminateCall]
+    errors: Optional[List[WhatsAppError]] = None
+
+
+class WhatsAppChange(BaseModel):
+    """Webhook change event wrapper.
+
+    Attributes:
+        value: The actual event data (connect or terminate)
+        field: Always "calls" for calling webhooks
+    """
+
+    value: Union[WhatsAppConnectCallValue, WhatsAppTerminateCallValue]
+    field: str
+
+
+class WhatsAppEntry(BaseModel):
+    """Webhook entry containing one or more changes.
+
+    Attributes:
+        id: WhatsApp Business Account ID
+        changes: List of change events in this webhook delivery
+    """
+
+    id: str
+    changes: List[WhatsAppChange]
+
+
+class WhatsAppWebhookRequest(BaseModel):
+    """Complete webhook request from WhatsApp.
+
+    This is the top-level structure for all webhook deliveries from
+    the WhatsApp Cloud API for calling events.
+
+    Attributes:
+        object: Always "whatsapp_business_account"
+        entry: List of webhook entries (usually contains one entry)
+    """
+
+    object: str
+    entry: List[WhatsAppEntry]
+
+
+class WhatsAppApi:
+    """WhatsApp Cloud API client for handling calls.
+
+    This class provides methods to interact with the WhatsApp Cloud API
+    for managing voice calls, including answering, rejecting, and terminating calls.
+
+    Attributes:
+        BASE_URL: Base URL for WhatsApp Graph API v23.0
+        phone_number_id: Your WhatsApp Business phone number ID
+        session: aiohttp client session for making HTTP requests
+        whatsapp_url: Complete URL for the calls endpoint
+        whatsapp_token: Bearer token for API authentication
+    """
+
+    BASE_URL = f"https://graph.facebook.com/v23.0/"
+
+    def __init__(
+        self, whatsapp_token: str, phone_number_id: str, session: aiohttp.ClientSession
+    ) -> None:
+        """Initialize the WhatsApp API client.
+
+        Args:
+            whatsapp_token: WhatsApp access token for authentication
+            phone_number_id: Your business phone number ID from WhatsApp Business API
+            session: aiohttp ClientSession for making HTTP requests
+        """
+        self.phone_number_id = phone_number_id
+        self.session = session
+        self.whatsapp_url = f"{self.BASE_URL}{phone_number_id}/calls"
+        self.whatsapp_token = whatsapp_token
```

Comment:
if these are internal we should prefix them with underscore.

Reply:
Done.